### PR TITLE
Add task enhancements

### PR DESCRIPTION
**Pull request by novakov-alexey**

### CODE REVIEW

File: cassandra/2-SubArrayWithMaxSum.cql [hunk 1]
```diff
@@ -10,7 +10,7 @@ INSERT INTO devgym.task (
 ) VALUES (
   '2016-01-01 00:00:00 UTC',
   'scalaClass',
-  now(),
+  9894cd10-ce12-11e5-8ee9-091830ac5256,
```
Comment (DmytroOrlov) [inline cassandra/2-SubArrayWithMaxSum.cql]: why uuid is better here?

File: server/app/controllers/AddTask.scala [hunk 2]
```diff
@@ -41,7 +42,7 @@ class AddTask @Inject()(executor: DynamicSuiteExecutor, dao: Dao, val messagesAp
         val check = Future(StringBuilderRunner(executor(f.referenceSolution, f.suite))).check
         check.flatMap { _ =>
           dao.addTask(NewTask(scalaClass, f.name, f.description, f.solutionTemplate, f.referenceSolution, f.suite))
-            .map(_ => Redirect(routes.Application.index))
+            .map(_ => Redirect(routes.AddTask.getAddTask).flashing(flashToUser -> messagesApi(taskAdded)))
```
Comment (DmytroOrlov) [inline server/app/controllers/AddTask.scala]: it seems no test for this, but manual check passed :)

File: server/app/dal/Dao.scala [hunk 3]
```diff
@@ -89,7 +89,7 @@ object Dao {
 
   val now = 0
 
-  def year(ago: Int = now) =
+  def yearAsOfJan1(ago: Int = now) =
```
Comment (DmytroOrlov) [inline server/app/dal/Dao.scala]: oh, it's better

File: server/app/service/RunnerUtils.scala [hunk 4]
```diff
@@ -26,7 +26,7 @@ object WithSuiteException {
   def apply[B](msg: String)(block: => B): B =
     try block catch {
       case e: SuiteException => throw e
-      case NonFatal(e) => throw new SuiteException(msg + e.getMessage, Some(e))
+      case NonFatal(e) => throw new SuiteException(msg, Some(e))
```
Comment (DmytroOrlov) [inline server/app/service/RunnerUtils.scala]: we don't need e.getMessage? ok, this code is less verbose :)

File: server/app/service/ScalaDynamicRunner.scala [hunk 5]
```diff
@@ -18,9 +18,10 @@ trait ScalaDynamicRunner extends DynamicSuiteExecutor with DynamicExecution {
   def apply(solution: String, suite: String)
            (channel: String => Unit)
            (implicit s: Scheduler): Unit = {
-    val traitName = WithSuiteException("There is no trait type defined in the Test constructor, code: ") {
+    val traitName = WithSuiteException(s"There is no solution trait type defined in suite, code: $suite") {
       findTraitName(suite)
     }
+    classDefPattern.findFirstIn(solution).orElse(throw new SuiteException(s"There is no class definition in solution code: $solution"))
```
Comment (DmytroOrlov) [inline server/app/service/ScalaDynamicRunner.scala]: did we lose this code during refactoring?

File: cassandra/2-SubArrayWithMaxSum.cql (hunk 1 above)
Comment (novakov-alexey) [inline cassandra/2-SubArrayWithMaxSum.cql]: Used in ApplicationTest.scala

val Some(result) = route(FakeRequest(GET, s"/task/scalaClass/${Date.from(instant).getTime}/9894cd10-ce12-11e5-8ee9-091830ac5256"))

File: server/app/service/RunnerUtils.scala (hunk 4 above)
Comment (novakov-alexey) [inline server/app/service/RunnerUtils.scala]: No, it was useless crap in e.getMessage

File: server/app/service/ScalaDynamicRunner.scala (hunk 5 above)
Comment (novakov-alexey) [inline server/app/service/ScalaDynamicRunner.scala]: No sure.

**Pull request merged**